core/ir/encode_api.h

```diff
@@ -51,38 +51,52 @@ typedef enum _dr_isa_mode_t {
     DR_ISA_ARM_THUMB,         /**< Thumb (ARM T32). */
     DR_ISA_ARM_A64,           /**< ARM A64 (AArch64). */
     DR_ISA_RV64IMAFDC,        /**< RISC-V (rv64imafdc). */
+                              /**
+                               * This is a synthetic ISA that has the purpose of
+                               * preserving register dependencies and giving hints on the
+                               * type of operations each instruction is performing.
+                               *
+                               * For this reason some operations that work on instructions
+                               * coming from an actual ISA (e.g., #DR_ISA_AMD64) are not
+                               * supported.
+                               *
+                               * Currently we support:
+                               * - instr_convert_to_isa_regdeps(), which converts an
+                               *   #instr_t of an actual ISA to a #DR_ISA_REGDEPS
+                               *   instruction.
+                               * - instr_encode() and instr_encode_to_copy(), to encode a
```

Comment:
The other comment got lost here, repeating to get it next to the code: please explicitly state that non-encode-or-decode instr_*, and all opnd_*, routines all work as otherwise it sounds like they don't as they are not listed here.  I don't think all readers are going to take the list of instr_t components below to imply that instr_t operations are supported -- esp when this list here says it's all that we support.

Reply:
Oh sorry, made that explicit.